https://github.com/andk/cpanpm/pull/175

Patch can be removed once we start shipping a perl release with this fix.

--- perl-5.36.0/cpan/CPAN/lib/CPAN/HTTP/Client.pm
+++ perl-5.36.0/cpan/CPAN/lib/CPAN/HTTP/Client.pm
@@ -32,6 +32,7 @@ sub mirror {
 
     my $want_proxy = $self->_want_proxy($uri);
     my $http = HTTP::Tiny->new(
+        verify_SSL => 1,
         $want_proxy ? (proxy => $self->{proxy}) : ()
     );
 
